crates/core/tedge/src/cli/mqtt/subscribe.rs

```diff
@@ -122,25 +111,19 @@ fn subscribe(cmd: &MqttSubscribeCommand) -> Result<(), anyhow::Error> {
                         }
                         n_packets += 1;
                         if matches!(cmd.count, Some(count) if count > 0 && n_packets >= count) {
-                            eprintln!("INFO: Break");
+                            eprintln!("INFO: Received {n_packets} messages");
                             break;
                         }
                     }
                     Err(err) => error!("{err}"),
                 }
             }
-            Ok(Event::Outgoing(Outgoing::PingReq)) => {
-                if matches!(cmd.duration, Some(duration) if !duration.is_zero() && started.elapsed() > duration)
-                {
-                    eprintln!("INFO: Timeout");
-                    break;
-                }
-            }
             Ok(Event::Incoming(Incoming::Disconnect)) => {
                 eprintln!("INFO: Disconnected");
                 break;
             }
             Ok(Event::Outgoing(Outgoing::Disconnect)) => {
+                eprintln!("INFO: Interrupted");
                 break;
```

Comment:
Is the `INFO: Interrupted` message really needed here? When using `--duration` it seems to print out both "Timeout" and "Interrupted" (which is a bit noisy from a UX point of view).

```sh
$ tedge mqtt sub '#' --duration 100ms >/dev/null
INFO: Connected
INFO: Timeout
INFO: Interrupted
```

In contrast to use the `--count`, it only prints out the expected message after fulfilling the exit condition.

```sh
$ tedge mqtt sub '#' --duration 100ms --count 1 >/dev/null
INFO: Connected
INFO: Received 1 messages
```

So the differences in log messages is a bit unexpected.

Reply:
Output looks great now, thanks.